File: BasicAgentCode/Python/Agent.py
import asyncio
import os
from dotenv import load_dotenv
from llama_index.llms.google_genai import GoogleGenAI

# === Load environment variables ===
load_dotenv()
os.environ["GOOGLE_API_KEY"] = os.getenv("GOOGLE_API_KEY")

# === Create Gemini LLM ===
llm = GoogleGenAI(
    model="models/gemini-1.5-flash",
    api_key=os.environ["GOOGLE_API_KEY"],
)

# === Main Event Loop ===
async def main():
    print("Gemini Agent Ready! Type 'exit' to quit.\n")
    while True:
        user_input = input("You > ")
        if user_input.strip().lower() in {"exit", "quit"}:
            break
        try:
            response = await llm.acomplete(user_input)
            print("Agent >", str(response))
        except Exception as e:
            print(f"Error: {e}")

# === Entrypoint ===
if __name__ == "__main__":
    asyncio.run(main())

File: BasicAgentCode/Typescript/Agent.ts
import "dotenv/config";
import { agent } from "@llamaindex/workflow";
import { Settings } from "llamaindex";
import { gemini, GEMINI_MODEL } from "@llamaindex/google";

async function main() {
  // Set global LLM (optional, but useful for consistency)
  Settings.llm = gemini({
    apiKey: process.env.GOOGLE_API_KEY,
    model: GEMINI_MODEL.GEMINI_2_0_FLASH,
  });

  // Create a minimal agent without tools
  const myAgent = agent({
    description: "this is an agent", // this is not setting the role of the llm -> ask sir kashif about it
  });

  let prompt: string = "who are you?";
  // Run the agent
  const response = await myAgent.run(prompt);
  console.log(response.data.message.content);
}

main();
